contrib/endpoints/src/api_manager/context/service_context.cc

```diff
@@ -93,59 +60,18 @@ MethodCallInfo ServiceContext::GetMethodCallInfo(
 }
 
 const std::string& ServiceContext::project_id() const {
-  if (gce_metadata_.has_valid_data() && !gce_metadata_.project_id().empty()) {
-    return gce_metadata_.project_id();
+  if (!global_context_->project_id().empty()) {
+    return global_context_->project_id();
   } else {
     return config_->service().producer_project_id();
   }
 }
 
 std::unique_ptr<service_control::Interface> ServiceContext::CreateInterface() {
   return std::unique_ptr<service_control::Interface>(
-      service_control::Aggregated::Create(config_->service(),
-                                          config_->server_config(), env_.get(),
-                                          &service_account_token_));
-}
-
-std::unique_ptr<cloud_trace::Aggregator>
-ServiceContext::CreateCloudTraceAggregator() {
-  // If force_disable is set in server config, completely disable tracing.
-  if (config_->server_config() &&
-      config_->server_config()->cloud_tracing_config().force_disable()) {
-    env()->LogInfo(
-        "Cloud Trace is force disabled. There will be no trace written.");
-    return std::unique_ptr<cloud_trace::Aggregator>();
-  }
-
-  std::string url = kCloudTraceUrl;
-  int aggregate_time_millisec = kDefaultAggregateTimeMillisec;
-  int cache_max_size = kDefaultTraceCacheMaxSize;
-  double minimum_qps = kDefaultTraceSampleQps;
-  if (config_->server_config() &&
-      config_->server_config()->has_cloud_tracing_config()) {
-    // If url_override is set in server config, use it to query Cloud Trace.
-    const auto& tracing_config =
-        config_->server_config()->cloud_tracing_config();
-    if (!tracing_config.url_override().empty()) {
-      url = tracing_config.url_override();
-    }
-
-    // If aggregation config is set, take the values from it.
-    if (tracing_config.has_aggregation_config()) {
-      aggregate_time_millisec =
-          tracing_config.aggregation_config().time_millisec();
-      cache_max_size = tracing_config.aggregation_config().cache_max_size();
-    }
-
-    // If sampling config is set, take the values from it.
-    if (tracing_config.has_samling_config()) {
-      minimum_qps = tracing_config.samling_config().minimum_qps();
-    }
-  }
-
-  return std::unique_ptr<cloud_trace::Aggregator>(new cloud_trace::Aggregator(
-      &service_account_token_, url, aggregate_time_millisec, cache_max_size,
-      minimum_qps, env_.get()));
+      service_control::Aggregated::Create(
+          config_->service(), global_context_->server_config().get(), env(),
```

Comment:
To be exactly,  service_control/* can depend on api_manager/*   but not api_manager/context/*